includes/DataObjects/GeoLocation.php

```diff
@@ -37,7 +41,7 @@ public function save()
 	{
 		if ($this->isNew) {
 // insert
-			$statement = $this->dbObject->prepare("INSERT INTO `geolocation` (address, data) VALUES (:address, :data);");
+			$statement = $this->dbObject->prepare("INSERT INTO `geolocation` (address, data) VALUES (:address, :data) ON DUPLICATE KEY UPDATE address = address;");
```

Comment:
What's the purpose of `ON DUPLICATE KEY UPDATE address = address` here?  Is your intention just to no-op instead of throwing an error if there's already an address in the table?

Reply:
Yep, exactly.

Reply:
Alright, fair enough